app/controllers/comments_controller.rb

```diff
@@ -1,17 +1,28 @@
 class CommentsController < ApplicationController
-  def create
-    @post = Post.find(params[:post_id])
-    @comment = @post.comments.build(comment_params)
-    if @comment.save
-      redirect_to post_path(@post), notice: 'Comment was successfully created.'
-    else
-      redirect_to post_path(@post), alert: 'Failed to create comment.'
-    end
+  def index
+    redirect_to user_posts_path
   end
 
-  private
+  def new
+    comment = Comment.new
+    respond_to do |format|
+      format.html { render :new, locals: { comment: } }
+    end
+  end
 
-  def comment_params
-    params.require(:comment).permit(:text)
+  def create
+    @current_user = current_user
+    comment = Comment.new(params.require(:comment).permit(:post_id, :text).merge(author_id: @current_user.id))
```

Comment:
- [ OPTIONAL] 
You could add a `private` params method to implement strong parameters. You can visit this [link](`https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides`) to read more about strong parameters.

**Example**
```ruby
  private 
  def comment_params
    params.require(:comment).permit(:text)
  end
```